crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -45,26 +46,26 @@ impl Persistence {
     pub async fn replace_current_auction(
         &self,
         auction: &domain::Auction,
-    ) -> Result<domain::auction::Id, Error> {
+    ) -> Result<domain::auction::Id, DatabaseError> {
         let auction = dto::auction::from_domain(auction.clone());
-        self.postgres
+        Ok(self
+            .postgres
             .replace_current_auction(&auction)
             .await
             .map(|auction_id| {
                 self.archive_auction(auction_id, auction);
                 auction_id
-            })
-            .map_err(Error::DbError)
+            })?)
```

Comment:
This needs to be consistent with other methods. Some of them use `map_err(DatabaseError)` instead. Since `impl From<sqlx::Error> for DatabaseError` was added, we should replace it everywhere with either `Ok(...?)` or `map_err(Into::into)` pattern(I personally find the second option a bit more readable).